barretenberg/cpp/src/barretenberg/dsl/acir_proofs/major-todos.md

```diff
@@ -0,0 +1,36 @@
+Major TODOS
+acir_format::Composer (and other types) become GUH
+? stretch update create_circuit to use Goblin gates (build_contraints and circuit_buf_to_acir_format in particular?)
+Update AcirComposer
+ - Wrap a GUH builder
+ - ? update create_circuit to populate that builder
+ - Wrap a GUH composer
+ - Wrap a Goblin
+ - ? init_proving_key 
+ - create_proof proxies to goblin.create_proof that we wrote
+ - proving and verification keys become GUH keys
+ - Serialization of proof 
+
+ Update Goblin: 
+  - Flesh out what we already did
+  - Add constructor from a GUH pk-vk pair (maybe no pk needed?)
+
+  Make acir_format and acir_proofs tests work
+
+  Stetch: try to understand what is_recursive flag should do
```

Comment:
We allow two proving systems to be created, once of them is "snark friendly" meaning that it uses PedersenBlake in the transcript and the other is solidity friendly meaning that it uses keccak in the transcript.

When creating a proof that we will want to verify in a another circuit, we use the is_recursive flag. 

See https://hackmd.io/@6iQDuIePQjyYBqDChYw_jg/SyLHcKsSa for a few more details

Reply:
Thanks for explaining. Our plans is to do violence to the code to get something working in the non-recursive case this week then check in with you and Maxim (prob next year?) to look things over together and rework to something non-disgusting :)

Reply:
Oh god :)

Reply:
Bear in mind that if a release is made, then this code will make its way into Noir user's hands -- haven't looked over the changes, though it seems that the solidity verifier contract may no longer work for them if this were to be merged?

Reply:
Ok don't worry, we just want to get it working in this branch, won't merge